### PR TITLE
feat: Refresh Configuration API PAGOPA-653

### PR DESCRIPTION
**Pull request by cap-ang**
#### List of Changes

- `GET Refresh Configuration activation` Global
- `GET Refresh Configuration activation` by Domain
- `GET Job Trigger activation` by Job type

#### Motivation and Context

To make it possible to refresh node configurations, that is, to trigger the loading and instances update in memory.

#### How Has This Been Tested?

- Unit testing
- manual testing (front-end vs back-end)

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/resources/application-h2.properties [hunk 1]
```diff
@@ -49,3 +49,8 @@ service.marketplace.subscriptionKey=subscriptionKey
 service.utils.host=http://localhost:8586
 service.utils.subscriptionKey=subscriptionKey
 
+# Trigger Job configuration
+service.trigger.job.host=${TRIGGER_JOB_HOST}
+
+# Refresh Configuration per domains configuration
+service.refresh-config.domains.host=${REFRESH_CONFIG_HOST}
```
Comment (jacopocarlini) [inline src/main/resources/application-h2.properties:56]:
```suggestion

# Trigger Job configuration
service.trigger.job.host=http://localhost:8587

# Refresh Configuration per domains configuration
service.refresh-config.domains.host=http://localhost:8588
```

File: src/test/java/it/pagopa/pagopa/apiconfig/service/RefreshServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.web.client.RestTemplate;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.matches;
+import static org.mockito.Mockito.when;
+
+@TestPropertySource(properties = {
```
Comment (jacopocarlini) [inline src/test/java/it/pagopa/pagopa/apiconfig/service/RefreshServiceTest.java:24]: you can add these properties in the file `src/test/resources/application.properties`

File: src/main/java/it/pagopa/pagopa/apiconfig/service/RefreshService.java [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+@Service
+@Validated
+public class RefreshService {
+    public enum ConfigDomain {
+        FTP_SERVER, INFORMATIVA_CDI, INFORMATIVA_PA, PA, PDD, PSP, Global
+    }
+
+    public enum JobTrigger {
+        paInviaRt, paRetryPaInviaRtNegative, paInviaRtRecovery, paSendRt
+    }
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/pagopa/apiconfig/service/RefreshService.java:26]: put these enumerations in separated file (`model` folder)

File: src/main/java/it/pagopa/pagopa/apiconfig/service/RefreshService.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    public String refreshConfig(ConfigDomain configType) {
+        String url;
+        if(configType.equals(ConfigDomain.Global))
+          url = jobTriggerUrl + "/refreshConfiguration";
+        else url = refreshConfigUrl + "/" + configType;
+
+        return get(url).getBody();
+    }
+
+    private ResponseEntity<String> get(String url) {
```
Comment (jacopocarlini) [inline src/main/java/it/pagopa/pagopa/apiconfig/service/RefreshService.java:50]: I don't like this solution. Is it possible to use [FeignClient](https://github.com/pagopa/pagopa-afm-utils/blob/031435fd4b9a157572fa040cb90bed6580456e9e/src/main/java/it/gov/pagopa/afm/utils/service/MarketPlaceClient.java#L17)? If it's not possibile, I'd like to put this method in an utility class

**Review by jacopocarlini — COMMENTED** [review]
Great Work 💯 
Just some enhancements

File: src/main/java/it/pagopa/pagopa/apiconfig/service/RefreshService.java (hunk 4 above)
Comment (AngeloCaporaso) [reply src/main/java/it/pagopa/pagopa/apiconfig/service/RefreshService.java:50]: Yes, thanks for the hint

**Review by jacopocarlini — APPROVED** [review]
Top 🔝

**Review by jacopocarlini — APPROVED** [review]
Good job! 💪

**Pull request merged**